### PR TITLE
SSCS-10853 Hearing Postponed Event Change

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10853

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ss-evoco — DISMISSED** [review]

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/handlers/postpone/PostponeHearingHandlerTest.java [hunk 1]
```diff
@@ -1,42 +1,56 @@
...
 
 @RunWith(SpringRunner.class)
 @TestPropertySource(locations = "classpath:config/application_e2e.properties")
 @SpringBootTest
 public class PostponeHearingHandlerTest extends BaseHandler {
 
+    @Autowired
+    private ObjectMapper mapper;
+
+    @DisplayName("When a hearing postpone callback is made, the correct fields are changed")
```
Comment (filippo-possenti-version1) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/handlers/postpone/PostponeHearingHandlerTest.java:32]: Vague description. What are the "correct fields"?

Might want to split this test into multiple ones, one for each field you want to check.

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java [hunk 2]
```diff
@@ -272,6 +272,10 @@ public void givenNonSAndLCase_shouldNoMessages() {
         PreSubmitCallbackResponse<SscsCaseData> response =
             handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
 
+        assertThat(response.getErrors())
+            .hasSize(1)
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java:276]: Question: do we care that there is only one error? Or we just want to ensure that there is "at least" the error we are looking for?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postpone/PostponeHearingHandlerTest.java [hunk 3]
```diff
@@ -1,73 +1,243 @@
...
+        + "but returns the correct error")
+    @Test
+    public void handleInvalidPostponementRelistEventType() {
+        caseData.getPostponement().setPostponementEvent(APPEAL_RECEIVED);
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+            handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getErrors())
+            .hasSize(1)
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postpone/PostponeHearingHandlerTest.java:188]: Same as above: do we care that there's exactly one error?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postpone/PostponeHearingHandlerTest.java [hunk 4]
```diff
@@ -1,73 +1,243 @@
...
+        + "or call update ccd but returns the correct error")
+    @Test
+    public void handleAlreadyHandled() {
+        caseData.getPostponement().setUnprocessedPostponement(NO);
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler
+            .handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getErrors())
+            .hasSize(1)
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postpone/PostponeHearingHandlerTest.java:209]: Same as above: do we care that there's exactly one error?

**Review by filippo-possenti-version1 — APPROVED** [review]
Nothing obvious to me. As long as it's tested, it's good to go in my view.

**Review by amoslv1 — APPROVED** [review]

**Review by amoslv1 — DISMISSED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Pull request merged**